src/app/cli/src/init/graphql_internal.ml

```diff
@@ -235,51 +235,106 @@ struct
+        in
+        match auth_keys with
+        | None ->
+            respond ()
+        | Some pks -> (
+            (* auth_keys is Some iff authentication is required
+               only caller is `create_graphql_server`, which enforces that invariant
+            *)
+            match Cohttp.Header.get headers "Authorization" with
+            | None ->
+                unauthorized ()
+            | Some auth -> (
+                let open Core in
+                let verify_signature_and_respond pk signature =
+                  (* we need to know not only that the pk verifies the
+                     signature, but also that this pk is allowed to verify
+                     the signature; otherwise, the client could sign a
+                     request using an arbitrary pk
                   *)
-                  match Cohttp.Header.get headers "Authorization" with
-                  | None ->
-                      unauthorized ()
-                  | Some auth -> (
-                      let open Core in
-                      match String.split_on_chars auth ~on:[ ' ' ] with
-                      | [ "Signature"; pk_str; signature ] -> (
-                          match Itn_crypto.pubkey_of_base64 pk_str with
-                          | Ok pk ->
-                              (* we need to know not only that the pk verifies the
-                                 signature, but also that this pk is allowed to verify
-                                 the signature; otherwise, the client could sign a
-                                 request using an arbitrary pk
-                              *)
-                              if
-                                List.mem pks pk ~equal:Itn_crypto.equal_pubkeys
-                                && Itn_crypto.verify ~key:pk ~msg:body_str
-                                     signature
-                              then respond ()
-                              else unauthorized ()
-                          | Error _err ->
-                              (* not a base64-encoded key *)
-                              unauthorized () )
-                      | _ ->
-                          (* invalid auth scheme *)
-                          unauthorized () ) ) ) )
+                  if
+                    List.mem pks pk ~equal:Itn_crypto.equal_pubkeys
+                    && Itn_crypto.verify ~key:pk ~msg:body_str signature
+                  then respond ()
+                  else unauthorized ()
+                in
+                match String.split_on_chars auth ~on:[ ' ' ] with
+                | [ "Signature"
+                  ; pk_str
+                  ; signature
+                  ; ";"
+                  ; "Sequencing"
+                  ; uuid_str
+                  ; seq_no_str
+                  ] -> (
+                    let pk_or_error = Itn_crypto.pubkey_of_base64 pk_str in
+                    let uuid_or_error =
```

Comment:
Nit: maybe it would be easier to just compare `Itn_sequencing.uuid` (which we can keep as string) to whatever string is provided here.

It would simplify code here and of `valid_sequence_number`.

Reply:
I want to distinguish improperly-formatted Uuid errors from not-the-correct Uuid errors.

That's more informative to the user.